### ISSUE TITLE
Can't play NPR stream

### ISSUE THREAD

**Issue by brownjd** [mopidy/mopidy#1697]
I get the following error when trying to play this stream: 

https://ondemand.npr.org/npr-mp4/npr/newscasts/2018/08/23/newscastShort100806.mp4

ERROR    GStreamer error: gst-stream-error-quark: Internal data stream error. (1)

I tested with it with the following command line options (I do have output = !qtdemux ! faad ! autoaudiosink in mopidy.conf) : 

gst-launch-1.0 https://ondemand.npr.org/npr-mp4/npr/newscasts/2018/08/23/newscastShort100806.mp4 ! qtdemux ! faad ! autoaudiosink
Setting pipeline to PAUSED ...
Pipeline is PREROLLING ...
Redistribute latency...
Pipeline is PREROLLED ...
Setting pipeline to PLAYING ...
New clock: GstPulseSinkClock
^Chandling interrupt.
Interrupt: Stopping pipeline ...
Execution ended after 0:00:06.971917034
Setting pipeline to PAUSED ...
Setting pipeline to READY ...
Setting pipeline to NULL ...
Freeing pipeline ...

**Comment by brownjd** [comment]
How can I get mopidy to log the gst-launch command it is using at runtime?

**Comment by brownjd** [comment]
Unfortunately, Mopidy does not respect GST_DEBUG_DUMP_DOT_DIR=/tmp mopidy

My guess is that it's not getting to that point. It would be really helpful if mopidy would print out the GStreamer pipeline.

**Comment by kingosticks** [comment]
Are you sure about that, the documentation explicitly mentions that's possible (https://docs.mopidy.com/en/latest/troubleshooting/).

**Comment by brownjd** [comment]
After running from source and putting in a bunch of debug statements, I got it to work. The problem was an early return in audio/actor.py

if new_state == Gst.State.READY:
   return  # Ignore READY state as it's GStreamer specific

It never made it far enough to dump the dot file. Once I moved it the lines that were at the end of on_playbin_state_changed to before the that return, it worked great and I was able to debug my pipeline.

Also, as an aside setup.py develop never worked (I kept getting old copies despite cleaning out dust, build, and python2.7/dist and site. I had to run setup.py install after every code change.

This issue can be closed, but the dot file generation probably needs a issue.
